src/unix/bsd/freebsdlike/freebsd/mod.rs

```diff
@@ -1720,6 +1723,72 @@ s_no_extra_traits! {
         pub uc_flags: c_int,
         __spare__: [c_int; 4],
     }
+
+    pub struct xinpgen {
+        pub xig_len: ksize_t,
+        pub xig_count: u32,
+        pub _xig_spare32: u32,
+        pub xig_gen: inp_gen_t,
+        pub xig_sogen: so_gen_t,
+        pub _xig_spare64: [u64; 4],
```

Comment:
The convention used elsewhere in libc is to make spare and padding fields private.  The forces consumers to initialize with `mem::zeroed`, but it also makes it easier for future versions of libc to replace those spare fields with no backwards-compatibility problems.
```suggestion
        _xig_spare32: u32,
        pub xig_gen: inp_gen_t,
        pub xig_sogen: so_gen_t,
        _xig_spare64: [u64; 4],
```